src/alchemlyb/tests/test_convergence.py

```diff
@@ -26,6 +32,85 @@ def test_convergence_fep(gmx_benzene_Coulomb_u_nk, estimator):
     assert convergence.loc[9, "Backward"] == pytest.approx(3.04, 0.01)
 
 
+def test_block_average_ti(gmx_benzene_Coulomb_dHdl):
+    print(type(gmx_benzene_Coulomb_dHdl), type(gmx_benzene_Coulomb_dHdl[0]))
+    df_avg = block_average(gmx_benzene_Coulomb_dHdl, "TI")
+    assert df_avg.shape == (9, 2)
+    assert df_avg.loc[1, "FE"] == pytest.approx(3.18, 0.01)
+    assert df_avg.loc[1, "FE_Error"] == pytest.approx(0.07, 0.1)
+    assert df_avg.loc[8, "FE"] == pytest.approx(3.15, 0.01)
+    assert df_avg.loc[8, "FE_Error"] == pytest.approx(0.07, 0.1)
+
+
+@pytest.mark.parametrize("estimator", ["DUMMY"])
+def test_block_average_error_1(gmx_ABFE_complex_u_nk, estimator):
+    with pytest.raises(ValueError, match=r"Estimator DUMMY is not available .*"):
+        _ = block_average(gmx_ABFE_complex_u_nk, estimator)
+
+
+@pytest.mark.parametrize("estimator", ["MBAR"])
+def test_block_average_error_2_mbar(gmx_ABFE_complex_u_nk, estimator):
+    df_list = gmx_ABFE_complex_u_nk[10:15]
```

Comment:
I kind of prefer that we would support this dataset.

Reply:
The result of running that would not be meaningful since the lambda state changes over the course of a trajectory.

Reply:
Notice that if this test ran as `_ = block_average(df_list, estimator)` it would pass the way you want it to. I'm concatenating the DataFrames to make a set where the lambda state changes internally just to throw the error.